code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -274,6 +283,8 @@
 		if(ranged) //We ranged? Shoot at em
 			if(!target.Adjacent(targets_from) && ranged_cooldown <= world.time) //But make sure they're not in range for a melee attack and our range attack is off cooldown
 				OpenFire(target)
+		//Component handles all the math and charge handling, so we can just send the signal here.
+		SEND_SIGNAL(src, COMSIG_HOSTILE_CHARGING_TARGET, target=target)
```

Comment:
I'd name this signal something else like try ON_TARGET_APPROACH or similar.

Reply:
![image](https://user-images.githubusercontent.com/20987591/91323474-327d5380-e7c1-11ea-92e0-ba9aaa1b510a.png)